packages/web-api/post-scans-func/index.ts

```diff
@@ -3,11 +3,11 @@
 import 'reflect-metadata';
 
 import { Context } from '@azure/functions';
-import { ScanController } from '../src/controllers/scan-controller';
+import { ControllerDispatcher } from '../src/controllerDispatcher';
+import { ScanRequestController } from '../src/controllers/scan-request-controller';
+import { setupIoContainer } from '../src/setupIoContainer';
 
 export async function run(context: Context): Promise<void> {
-    const controller = new ScanController(context);
-    if (controller.validateRequest()) {
-        controller.postScanRunRequests();
-    }
+    const dispatcher = new ControllerDispatcher(ScanRequestController, context, setupIoContainer());
```

Comment:
>ainer()); [](start = 91, length = 9)

may be pass the api name here for telemetry purpose

Reply:
app insight is attached to the func app and all requests should be logged by default